src/missingarray.jl

```diff
@@ -26,3 +26,13 @@ Base.fill!(A::CategoricalArray{>:Missing}, ::Missing) = (fill!(A.refs, 0); A)
 
 in(x::Missing, y::CategoricalArray) = false
 in(x::Missing, y::CategoricalArray{>:Missing}) = !all(v -> v > 0, y.refs)
+
+function Missings.replace(a::CategoricalArray{S, N, R, V, C}, replacement::V) where {S, N, R, V, C}
+    pool = deepcopy(a.pool)
+    v = C(get!(pool, replacement), pool)
+    Missings.replace(a, v)
+end
+
+function collect(r::Missings.EachReplaceMissing{CategoricalArray{S, N, R, C, V, U}}) where {S, N, R, C, V, U}
```

Comment:
You should be able to just do `<:CategoricalArray{S, N, R, C}`. Below, it would be useful to add `R[...]` in front of the comprehension for empty arrays, just in case.

Reply:
Ah, I didn't know you could use <: like that. Nice

It seems to me that `[v.level for v in r]` could deduce the array type for empty arrays as well. But I added `R[...]` because why not.